ledger-core/virtual/execute/execute.go

```diff
@@ -678,7 +680,11 @@ func (s *SMExecute) stepSendOutgoing(ctx smachine.ExecutionContext) smachine.Sta
 			return ctx.Error(throw.E("failed to publish bargeInCallback"))
 		}
 	} else {
-		s.outgoing.CallRequestFlags = payload.BuildCallRequestFlags(payload.SendResultDefault, payload.RepeatedCall)
+		if s.outgoingSentCounter < MaxOutgoingSendCount {
+			s.outgoing.CallRequestFlags = payload.BuildCallRequestFlags(payload.SendResultDefault, payload.RepeatedCall)
+		} else {
+			return ctx.Error(throw.NotImplemented())
```

Comment:
It's never gonna be implemented, please change to something like throw.New("outgoing retries limit"), so we could track the error in logs

Reply:
+1

Reply:
ok
fun fact - throw.New or throw.E doesn't have any information about SM step